components/backup-stream/src/endpoint.rs

```diff
@@ -208,10 +221,33 @@ where
     RT: RaftStoreRouter<E> + 'static,
     PDC: PdClient + 'static,
 {
-    fn on_fatal_error(&self, _task: String, _err: Box<Error>) {
-        // This is a stub.
-        // TODO: implement the feature of reporting fatal error to the meta storage,
-        //       and pause the task then.
+    fn get_meta_client(&self) -> MetadataClient<S> {
+        self.meta_client.as_ref().unwrap().clone()
+    }
+
+    fn on_fatal_error(&self, task: String, err: Box<Error>) {
+        // Let's pause the task locally first.
+        self.on_unregister(&task);
+
+        let meta_cli = self.get_meta_client();
+        let store_id = self.store_id;
+        let sched = self.scheduler.clone();
+        self.pool.block_on(async move {
+            // TODO: also pause the task using the meta client.
+            let mut last_error = StreamBackupError::new();
+            last_error.set_error_code(err.error_code().code.to_owned());
+            last_error.set_error_message(err.to_string());
+            last_error.set_store_id(store_id);
+            last_error.set_happen_at(TimeStamp::physical_now());
+            if let Err(err_report) = meta_cli.report_last_error(&task, last_error).await {
+                err_report.report(format_args!("failed to upload error {}", err_report));
+                // Let's retry reporting after 5s.
```

Comment:
why keep reporting this error every 5s?🤔

Reply:
If failed to report, we would retry report error by this. (Just a rough retry implement)